nvflare/app_common/streamers/file_streamer.py

```diff
@@ -178,7 +183,13 @@ def register_stream_processing(
         if not os.path.isdir(dest_dir):
             raise ValueError(f"dest_dir '{dest_dir}' is not a valid dir")
 
-        engine = fl_ctx.get_engine()
+        if isinstance(fl_ctx, StreamableEngine):
```

Comment:
Should we allow this, passing engine as fl_ctx? I think it's better to throw an error here.